sdcm/utils/common.py

```diff
@@ -229,18 +229,36 @@ def generate_url(self, file_path, dest_dir=''):
 
     def upload_file(self, file_path, dest_dir=''):
         s3_url = self.generate_url(file_path, dest_dir)
+        s3_obj = "{}/{}".format(dest_dir, os.path.basename(file_path))
         try:
             LOGGER.info("Uploading '{file_path}' to {s3_url}".format(file_path=file_path, s3_url=s3_url))
             print "Uploading '{file_path}' to {s3_url}".format(file_path=file_path, s3_url=s3_url)
             self._bucket.upload_file(Filename=file_path,
-                                     Key="{}/{}".format(dest_dir, os.path.basename(file_path)),
+                                     Key=s3_obj,
                                      Config=self.transfer_config)
             LOGGER.info("Uploaded to {0}".format(s3_url))
+            LOGGER.info("Set public read access")
+            self.set_public_access(key=s3_obj)
             return s3_url
         except Exception as details:  # pylint: disable=broad-except
             LOGGER.debug("Unable to upload to S3: %s", details)
             return ""
 
+    def set_public_access(self, key):
+        import copy
```

Comment:
please move the import to the top of the file

Reply:
fixed